test/client.ts

```diff
@@ -615,6 +615,167 @@ describe('MqttClient', () => {
 		},
 	)
 
+	describe('connect manually', () => {
+		it(
+			'should not throw an error when publish after second connect',
+			{
+				timeout: 10000,
+			},
+			async function _test(t) {
+				const fail = await new Promise<boolean>((resolveParent) => {
+					let countConnects = 0
+					const clock = useFakeTimers({
+						shouldClearNativeTimers: true,
+						toFake: ['setTimeout'],
+					})
```

Comment:
Usually it would be better to do this inside `before()` and `after()` (to put inside the test fn)

Reply:
¿Open the clock in before y close it in after?

Reply:
See example here: 
https://github.com/mqttjs/MQTT.js/blob/cd0b044b89e8413e99e055529d523eac6d4ddcd4/test/abstract_client.ts#L2150-L2156

Sorry only after is needed here no before